docs/developer-docs/smart-contracts/overview/inside-canisters.mdx

```diff
@@ -0,0 +1,58 @@
+# Inside canisters
+
+Canisters are WebAssembly (Wasm) programs with additional features that allow them to store persistent data, communicate with users and other canisters, and be managed by DAOs.
+This means that in addition to the standard Wasm components such as Wasm code and Wasm memory, canisters also have other components that are explained in the rest of this page.
+
+<div class="text--center">
+<img src="/img/docs/inside-canister.png" alt="Canister" width="600"/>
+</div>
+
+## Canister Id
+A canister has a unique identifier that is often referred to as the address of the canister or the principal of the canister.
+When a user or a canister sends a message to another canister, they specify the target canister by its id in the header of the message.
+The system then routes the message to the target canister and puts it in the input queue of that canister.
+
+## Input queue
+Canisters have queues for incoming messages.
+When a canister is scheduled for execution, it takes one message from the queue and runs the corresponding Wasm function specified by the message.
+The code of the function is stored in the Wasm binary of the canister.
+During execution, the function reads the payload of the message and modifies data stored in the canister.
+
+::infobox
+The order of messages in the queue depends on the implementation and it is not necessarily FIFO (first-in first-out).
+
+## Canister storage
+There are two [storages](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/maintain/storage) available to the canister: the Wasm memory and the stable memory.
+The function doesn’t need to do anything special to use the Wasm memory since this memory type  is automatically used for heap-allocated objects.
```

Comment:
```suggestion
The function doesn’t need to do anything special to use the Wasm memory since this memory type is automatically used for heap-allocated objects.
```